Source/WebKit/UIProcess/API/glib/WebKitWebExtension.cpp

```diff
@@ -0,0 +1,1298 @@
+        g_param_spec_boolean(
+            "has-commands",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    /**
+     * WebKitWebExtension:has-content-modification-rules:
+     * 
+     * Whether the #WebKitWebExtension includes rules used for content modification or blocking.
+     * See webkit_web_extension_get_content_modification_rules() for more details.
+     * 
+     * Since: 2.52
+     */
+    properties[PROP_HAS_CONTENT_MODIFICATION_RULES] =
+        g_param_spec_boolean(
+            "has-content-modification-rules",
+            nullptr, nullptr,
+            FALSE,
+            WEBKIT_PARAM_READABLE);
+
+    g_object_class_install_properties(objectClass, properties.size(), properties.data());
+}
+
+static gboolean webkitWebExtensionInitableInit(GInitable* initable, GCancellable* cancellable, GError** error)
+{
+#if ENABLE(WK_WEB_EXTENSIONS)
+    WebKitWebExtension* self = WEBKIT_WEB_EXTENSION(initable);
+    GRefPtr<GFile> extensionPath = self->priv->path;
+    g_return_val_if_fail(WEBKIT_IS_WEB_EXTENSION(self), FALSE);
+    g_return_val_if_fail(extensionPath && G_IS_FILE(extensionPath.get()), FALSE);
+    g_return_val_if_fail(g_file_query_file_type(extensionPath.get(), G_FILE_QUERY_INFO_NONE, nullptr) == G_FILE_TYPE_DIRECTORY, FALSE);
+
+    RefPtr<API::Error> internalError;
+    Ref extension = WebKit::WebExtension::create(extensionPath.get(), internalError);
+    if (internalError) {
+        g_set_error(error, webkit_web_extension_error_quark(),
+            toWebKitWebExtensionError(internalError->errorCode()), internalError->localizedDescription().utf8().data(), nullptr);
+        return FALSE;
+    }
+
+    self->priv->extension = WTFMove(extension);
+
+    return TRUE;
+#else
+    return FALSE;
+#endif
+}
+
+static void gInitableInterfaceInit(GInitableIface* iface)
+{
+    iface->init = webkitWebExtensionInitableInit;
+}
+
+#if ENABLE(WK_WEB_EXTENSIONS)
+
+WebKitWebExtension* webkitWebExtensionCreate(HashMap<String, GRefPtr<GBytes>>&& resources, GError** error)
+{
+    WebExtension::Resources newResources;
+    for (auto& it : resources)
+        newResources.add(it.key, API::Data::createWithoutCopying(WTFMove(it.value)));
+
+    Ref extension = WebKit::WebExtension::create(WTFMove(newResources));
+
+    if (!extension->errors().isEmpty()) {
+        Ref internalError = extension->errors().last();
+        g_set_error(error, webkit_web_extension_error_quark(),
+            toWebKitWebExtensionError(internalError->errorCode()), internalError->localizedDescription().utf8().data(), nullptr);
+    }
+
+    WebKitWebExtension* object = WEBKIT_WEB_EXTENSION(g_object_new(WEBKIT_TYPE_WEB_EXTENSION, nullptr));
+    object->priv->extension = WTFMove(extension);
+    return object;
+}
+
+/**
+ * webkit_web_extension_new:
+ * @extension_path: A #GFile pointing to the folder containing the extension manifest and resources
+ * @error: return location for error or %NULL to ignore
```

Comment:
So just to make sure you're aware, this is gtk-doc syntax. We should eventually convert it to gi-docgen syntax. But gtk-doc syntax is still widely-used throughout our API docs, and we use gi-docgen syntax almost nowhere, so you don't actually have to worry about this yourself.

Reply:
Yes, I am aware. I occasionally try and remember to use gi-docgen syntax, but since gtk-doc is so prominent throughout this codebase I sometimes forget.